### PR TITLE
Add {locked,total}_balance to Rosetta account/balance response

### LINKED ISSUE #8124: Rosetta account/balance should return total and locked balances
For example, like this
```json
{
    "block_identifier": {
        "index": 338,
        "hash": "3NLQungv8KTRuj7tQeBmPVYZHtvomh9pqB1b2MnHayJ3oh5aA5F8"
    },
    "balances": [
        {
            "value": "0",
            "currency": {
                "symbol": "CODA",
                "decimals": 9
            },
            "metadata": {
                "total_balance": "66000",
                "locked_balance": "66000"
            }
        }
    ],
    "metadata": {
        "nonce": 0
    }
}
```

### PR DESCRIPTION
**Pull request by omerzach**
Replaces responses like
```
"balances":[{"value":"7020000000000","currency":{"symbol":"CODA","decimals":9},"metadata":{}}]
``` 
with
```
"balances":[{"value":"7020000000000","currency":{"symbol":"CODA","decimals":9},"metadata":{"locked_balance":2980000000000,"total_balance":10000000000000}}]
```
which allows the client to see the locked and total balances and provides additional context for the existing liquid `value` field.

Tested changes by running the Rosetta integration tests, running `rosetta-cli`, and manually `curl`ing a local server.

Checklist:
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? List them:

Closes #8124

### CODE REVIEW

**Review by jspada — COMMENTED** [review]
LGTM, but this should be reviewed and approved by someone else who is more familiar.

I also wonder if this change must go into `develop` or whether it should go into `compatible` instead.

It only adds a new properties to `Amount.t`'s metadata, so anything using the balances should not be affected and only new code looking for the new fields would touch it.

**Review by bkase — COMMENTED** [review]
I agree with @jspada , this is backwards compatible right? It looks like it to me.

**Review by mrmr1993 — CHANGES_REQUESTED** [review]
Implementation looks good, but blocking on the design.

File: src/app/rosetta/lib/account.ml [hunk 1]
```diff
@@ -333,6 +358,24 @@ module Balance = struct
...
+          ( match token_id with
+          | None ->
+              Amount_of.coda
+          | Some token_id ->
+              Amount_of.token token_id )
+            liquid_balance
+        in
+        let metadata =
+          `Assoc
+            [ ( "liquid_balance"
```
[the remark this answers is not in the document]
Comment (jspada) [reply src/app/rosetta/lib/account.ml:372]: Right- good point. With `locked_balance` and `total_balance` is how it's requested in #8124 also. 👍🏼

File: src/app/rosetta/lib/account.ml [hunk 2]
```diff
@@ -339,6 +358,27 @@ module Balance = struct
...
+        let amount =
+          ( match token_id with
+          | None ->
+              Amount_of.coda
+          | Some token_id ->
+              Amount_of.token token_id )
+            liquid_balance
+        in
+        let locked_balance =
+          Unsigned.UInt64.sub total_balance liquid_balance
```
Comment (omerzach) [inline src/app/rosetta/lib/account.ml:371]: @mrmr1993 Is this fine for computing the locked balance or are there any gotchas I'm not thinking about?

File: src/app/rosetta/lib/account.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (omerzach) [reply src/app/rosetta/lib/account.ml]: Updated!

File: src/app/rosetta/lib/account.ml (hunk 2 above)
Comment (mrmr1993) [reply src/app/rosetta/lib/account.ml:371]: This is perfect, yep!

**Review by mrmr1993 — APPROVED** [review]
Great!

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by jspada — APPROVED** [review]

**Pull request merged**